tap_wikipedia/utils/wikipedia_abstracts_parser.py

```diff
@@ -1,87 +1,67 @@
-from pydantic import BaseModel
+from typing import Any
+from xml import sax
 
-import xml.sax as sax
-from typing import List, Dict, Tuple
-
-
-class _EntityXML(BaseModel):
-    """A Pydantic Model to contain Wikipedia Abstracts"""
-
-    class AbstractInfo(BaseModel):
-        title: str = ""
-        abstract: str = ""
-        url: str = ""
-
-    class Sublink(BaseModel):
-        anchor: str = ""
-        link: str = ""
-
-    abstract_info: AbstractInfo = AbstractInfo()
-    sublinks: List[Sublink] = []
+from tap_wikipedia.models import wikipedia
 
 
 class WikipediaAbstractsParser(sax.ContentHandler):
-    """SAX Handler for Wikipedia Abstracts"""
+    """SAX Handler for Wikipedia Abstracts."""
 
     def __init__(self):
-        self.currentData = ""
-        self.charBuffer: List[str] = []
-        self._records: List[Dict] = []
-        self.abstractData = _EntityXML()
+        self._records: list[wikipedia.Record] = []
+        self.abstractInfo: wikipedia.AbstractInfo | None
```

Comment:
The SAX callback methods (startElement, endElement, et al.) are named that way because the SAX spec came out of the Java universe. You should use Python conventions everywhere else.

Reply:
I am using the Python conventions in the parser now.